app/models/miq_region.rb

```diff
@@ -242,6 +243,45 @@ def remote_ws_url
     hostname.nil? ? nil : "https://#{hostname}"
   end
 
+  def generate_auth_key(ssh_user, ssh_password, ssh_host = remote_ws_address)
+    key = remote_region_v2_key(ssh_user, MiqPassword.try_decrypt(ssh_password), ssh_host)
+
+    auth = AuthToken.new
+    auth.auth_key = key
+    auth.name = "Region #{region} API Key"
+    auth.resource = self
+    auth.save!
+  end
+
+  def verify_credentials(_auth_type = nil, _options = nil)
+    # TODO: verify the key against the remote api using the api client gem
+    true
+  end
+
+  def api_system_auth_token(userid)
+    region_v2_key = authentication_token
+
+    token_hash = {
+      :server_guid => remote_ws_miq_server.guid,
+      :userid      => userid,
+      :timestamp   => Time.now.utc
+    }
+
+    file = Tempfile.new("region_auth_key")
+    begin
+      file.write(region_v2_key)
+      file.close
+      key = EzCrypto::Key.load(file.path)
+      MiqPassword.new.encrypt(token_hash.to_yaml, "v2", key)
```

Comment:
The class method form doesn't actually allow you to specify your own key.
See https://github.com/ManageIQ/manageiq/blob/master/gems/pending/util/miq-password.rb#L63 vs https://github.com/ManageIQ/manageiq/blob/master/gems/pending/util/miq-password.rb#L21